packages/ui/exchange-ui-core/src/containers/SupplyTable.tsx

```diff
@@ -0,0 +1,229 @@
+    });
+
+    useEffect(() => {
+        if (deviceClient) {
+            loadPage(1);
+        }
+    }, [deviceClient, supplySettings]);
+
+    useEffect(() => {
+        if (deviceClient) {
+            dispatch(fetchSupplies());
+        }
+    }, [deviceClient]);
+
+    const currentYear = moment().format('YYYY');
+    const nextYear = moment().add(1, 'years').format('YYYY');
+
+    const columns = [
+        { id: 'type', label: t('exchange.supply.type') },
+        { id: 'facility', label: t('exchange.supply.facilityName') },
+        { id: 'price', label: t('exchange.supply.price') },
+        { id: 'status', label: t('exchange.supply.status') },
+        {
+            id: 'certified',
+            label: `${t(
+                'exchange.supply.meterReadToBeCertified'
+            )} for ${currentYear}/${nextYear} (${EnergyFormatter.displayUnit})`
+        }
+    ] as const;
+
+    const rows = paginatedData.map(({ device }) => {
+        return {
+            type: device.deviceType?.replace(new RegExp(';', 'g'), ' - ') ?? '-',
+            facility: device.facilityName,
+            price: formatCurrencyComplete(device.price || 0 / 100, defaultCurrency),
+            status: device.active ? KeyStatus.Active : KeyStatus.Paused,
+            certified: EnergyFormatter.format(
+                BigNumber.from(device.meterStats?.uncertified ?? 0).toNumber()
+            )
+        };
+    });
+
+    const editSupply = (index: string) => {
+        const { device } = paginatedData[index];
+        setEntity({
+            ...device,
+            deviceType: device.deviceType?.replace(new RegExp(';', 'g'), ' - ') ?? '-',
+            defaultAskPrice: device.defaultAskPrice / 100
+        });
+        setShowModal(true);
+    };
+
+    const removeSupply = (index: string) => {
+        const { device } = paginatedData[index];
+        setToRemove(device);
+    };
+
+    const actions = [
+        {
+            icon: <Edit />,
+            name: t('exchange.supply.update'),
+            onClick: (idx: string) => editSupply(idx)
+        },
+        {
+            icon: <Remove />,
+            name: t('exchange.supply.remove'),
+            onClick: (idx: string) => removeSupply(idx)
+        }
+    ];
+
+    const STATUS_OPTIONS = Object.keys(KeyStatus).map((key) => ({
+        value: KeyStatus[key],
+        label: key.toString()
+    }));
+
+    const filters: ICustomFilterDefinition[] = [
+        {
+            property: (device: IDeviceWithSupply) => device.facilityName.toString(),
+            label: t('exchange.supply.facility'),
+            input: {
+                type: CustomFilterInputType.string
+            }
+        },
+        {
+            property: (device: IDeviceWithSupply) =>
+                device.active ? KeyStatus.Active : KeyStatus.Paused,
+            label: t('exchange.supply.status'),
+            input: {
+                type: CustomFilterInputType.dropdown,
+                availableOptions: STATUS_OPTIONS
```

Comment:
If you make `KeyStatus` and enum, you can remove `STATUS_OPTIONS` and only put `availableOptions : Object.values(KeyStatus)`